collect_app/src/main/java/org/odk/collect/android/widgets/utilities/GeoWidgetUtils.kt

```diff
@@ -0,0 +1,132 @@
+package org.odk.collect.android.widgets.utilities
+
+import android.content.Context
+import android.location.Location
+import org.odk.collect.android.R
+import org.odk.collect.maps.MapPoint
+import org.odk.collect.shared.strings.StringUtils.removeEnd
+import timber.log.Timber
+import java.text.DecimalFormat
+import kotlin.math.abs
+import kotlin.math.absoluteValue
+
+object GeoWidgetUtils {
+
+    @JvmStatic
+    fun getGeoPointAnswerToDisplay(context: Context, answer: String?): String {
+        try {
+            if (answer != null && answer.isNotEmpty()) {
+                val parts = answer.split(" ".toRegex()).toTypedArray()
+                return context.getString(
+                    R.string.gps_result,
+                    convertCoordinatesIntoDegreeFormat(context, parts[0].toDouble(), "lat"),
+                    convertCoordinatesIntoDegreeFormat(context, parts[1].toDouble(), "lon"),
+                    truncateDouble(parts[2]),
+                    truncateDouble(parts[3])
+                )
+            }
+        } catch (e: NumberFormatException) {
+            return ""
+        }
+
+        return ""
+    }
+
+    @JvmStatic
+    fun getGeoPolyAnswerToDisplay(answer: String?): String? {
+        return if (answer != null && answer.isNotEmpty()) {
+            removeEnd(answer.replace(";\\s".toRegex(), ";"), ";")
+        } else {
+            answer
+        }
+    }
+
+    @JvmStatic
+    fun parseGeometryPoint(answer: String?): DoubleArray? {
+        if (answer != null && answer.isNotEmpty()) {
+            val sa = answer.trim { it <= ' ' }.split(" ".toRegex()).toTypedArray()
+            return try {
+                DoubleArray(4).also {
+                    it[0] = sa[0].toDouble()
+                    it[1] = if (sa.size > 1) sa[1].toDouble() else 0.0
+                    it[2] = if (sa.size > 2) sa[2].toDouble() else 0.0
+                    it[3] = if (sa.size > 3) sa[3].toDouble() else 0.0
```

Comment:
It could be:

```
                doubleArrayOf(
                    sa[0].toDouble(),
                    if (sa.size > 1) sa[1].toDouble() else 0.0,
                    if (sa.size > 2) sa[2].toDouble() else 0.0,
                    if (sa.size > 3) sa[3].toDouble() else 0.0
                )
```